### DISCUSSION TITLE
Questions about the example codes of ElectrostaticPIC

### DISCUSSION THREAD

**Discussion by HoulinLee** [AMReX-Codes/amrex#2534] [Q&A] [answered]
Dear all,
I have some questions about the example codes of ElectrostaticPIC.
1. In the function `interpolate_cic_two_levels` at _amrex-tutorials-main/ExampleCodes/Particles/ElectrostaticPIC/src/particles/field_gather/FieldGather_K.H_,  the codes `if (lev == 1 && mask(i,j,0,0) == 1) {} else {}` indicate that the electric field of coarser grid is used near the level boundary when the current level is level 1. I wonder why we should gather the particle field in this way.
2. In _amrex-tutorials-main/ExampleCodes/Particles/ElectrostaticPIC/src/field_solver/FieldSolver.cpp_, `setLevelBC` is used in the function `computePhi`. I noticed that the multifabs `phi` and `rhs` are nodal types, but the comments of `setLevelBC` in _amrex/Src/LinearSolvers/MLMG/AMReX_MLLinOp.H_ say that `setLevelBC` is only for cell-centered solves only.
3. In the function `sum_fine_to_crse_nodal` at _amrex-tutorials-main/ExampleCodes/Particles/ElectrostaticPIC/src/field_solver/FieldSolver_K.H_, should `fine(ii-1,jj+1,0,0)` be `fine(ii+1,jj-1,0,0)`?

Thanks.

**Comment by atmyers** [comment, marked as answer]
1) This is done to mitigate self-force errors that occur when the macroparticle is near the coarse/fine boundaries - the method that this example code implements is described [here](https://www.cambridge.org/core/journals/laser-and-particle-beams/article/abs/mesh-refinement-for-particleincell-plasma-simulations-applications-to-and-benefits-for-heavy-ion-fusion/C1ACE61B0CEF032CAD7CFA8A53F76290). 

2) Yes, this function is a no-op in the nodal case. The call doesn't really need to be there, but it doesn't hurt anything to have it.

3) I believe you are correct, thank you for finding and reporting this. https://github.com/AMReX-Codes/amrex-tutorials/pull/46

**Reply by HoulinLee** [reply]
Thanks for your reply!
